src/main/java/io/kestra/plugin/gcp/runner/GcpBatchScriptRunner.java

```diff
@@ -0,0 +1,299 @@
+
+@Introspected
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(title = "Google Cloud Platform Batch script runner",
+    description = """
+        This job runner didn't resume the job if a Worker is restarted before the job finish.
+        You need to have roles 'Batch Job Editor' and 'Logs Viewer' to be able to use it.""")
+@Plugin(examples = {}, beta = true)
+public class GcpBatchScriptRunner extends ScriptRunner implements GcpInterface {
+    private static final String WORKING_DIR = "/kestra/working-dir";
+    private static final int BUFFER_SIZE = 8 * 1024;
+    public static final String MOUNT_PATH = "/mnt/disks/share";
+
+    private String projectId;
+    private String serviceAccount;
+    @Builder.Default
+    private List<String> scopes = Collections.singletonList("https://www.googleapis.com/auth/cloud-platform");
+
+    @Schema(
+        title = "The GCP region."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private String region;
+
+    @Schema(
+        title = "The GCP machine type.",
+        description = "See https://cloud.google.com/compute/docs/machine-types"
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @Builder.Default
+    private String machineType = "e2-medium";
+
+    @Schema(
+        title = "Compute reservation."
+    )
+    @PluginProperty(dynamic = true)
+    private String reservation;
+
+
+    @Schema(
+        title = "Google Cloud Storage Bucket to use to upload (`inputFiles` and `namespaceFiles`) and download (`outputFiles`) files.",
+        description = "It's mandatory to provide a bucket if you want to use such properties."
+    )
+    @PluginProperty(dynamic = true)
+    private String gcsBucket;
+
+    @Schema(
+        title = "The maximum duration to wait for the job completion. Google Cloud Batch will automatically timeout the job upon reaching such duration and the task will be failed."
+    )
+    @Builder.Default
+    private final Duration waitUntilCompletion = Duration.ofHours(1);
+
+    @Schema(
+        title = "Whether the job should be deleted upon completion."
+    )
+    @NotNull
+    @Builder.Default
+    private final Boolean delete = true;
+
+    @Override
+    public RunnerResult run(RunContext runContext, ScriptCommands commands, List<String> filesToUpload, List<String> filesToDownload) throws Exception {
+        String renderedBucket = runContext.render(gcsBucket);
+        String workingDirName = IdUtils.create();
+        Map<String, Object> additionalVars = commands.getAdditionalVars();
+        Optional.ofNullable(renderedBucket).ifPresent(bucket -> additionalVars.putAll(Map.<String, Object>of(
+            "gcsWorkingDir", "gs://" + bucket + "/" + workingDirName,
+            "workingDir", WORKING_DIR,
+            "outputDir", WORKING_DIR
+        )));
+
+        GoogleCredentials credentials = CredentialService.credentials(runContext, this);
+
+        // TODO internal storage files?
```

Comment:
You lack [this](https://github.com/kestra-io/plugin-azure/pull/79/files#diff-4429724fde116f257196e7f134f1a39a9b7c64545ef6b63871ffe2adfd7f509cR82-R87) I think